### PR TITLE
Show employees

### PR DESCRIPTION
**Pull request by gfontenot**
This adds some basic functionality to the app
- EmployeesViewController displays a list of employees
  - AsyncLoadingImageView lets us load images via URL. This is a super
    simple implementation. I plan on adding a cache later.
  - Extended Roles to be Printable
  - Added Gravatars to employees
  - Added a global `dispatch_main_async` function for performing blocks
    on the main thread. This is super duper common and it's nice to have
    it wrapped up. Not sure if this file is the best place for it.
  - Added a helper method to `NSURLSession` to clean up the API just a
    bit.

### CODE REVIEW

File: roster/Classes/ViewControllers/EmployeesViewController.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import UIKit
+
+class EmployeesViewController: UITableViewController {
+    var employees: [TBEmployee] = []
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        TBTeamClient().fetchAllEmployeesWithCompletion { (employees, error) -> Void in
```
Comment (tonyd256) [inline roster/Classes/ViewControllers/EmployeesViewController.swift]: I would do `{ employees, error in`

File: roster/Classes/ViewControllers/EmployeesViewController.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            self.tableView.reloadData()
+        }
+    }
+
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return self.employees.count
+    }
+
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = self.tableView.dequeueReusableCellWithIdentifier("EmployeeCell", forIndexPath: indexPath) as EmployeeCell
```
Comment (tonyd256) [inline roster/Classes/ViewControllers/EmployeesViewController.swift]: don't need `self` here

File: roster/Extensions/NSURLSession+FetchData.swift [hunk 3]
```diff
@@ -0,0 +1,9 @@
+extension NSURLSession {
+    class func fetchDataFromURL(url: NSURL, completionHandler: (NSData) -> ()) -> NSURLSessionDataTask {
+        return self.sharedSession().dataTaskWithURL(url) { data, _, _ in
+            if let actualData = data {
+                completionHandler(actualData)
+            }
```
Comment (tonyd256) [inline roster/Extensions/NSURLSession+FetchData.swift:6]: maybe have a default case here or the completion will never be called

File: roster/Extensions/TBEmployee+Gravatar.swift [hunk 4]
```diff
@@ -0,0 +1,5 @@
+extension TBEmployee {
+    var gravatarURL: NSURL {
+        return KHGravatar.URLForGravatarEmailAddress(emailAddress, defaultImageType:.ImageIdenticon, forceDefault: false, rating: .R, size: 100.0)
+    }
+}
```
Comment (tonyd256) [inline roster/Extensions/TBEmployee+Gravatar.swift]: new line

**Comment by tonyd256** [comment]
some comments but :+1:  once you fix them or ignore them

File: roster/Extensions/NSURLSession+FetchData.swift (hunk 3 above)
Comment (gfontenot) [inline roster/Extensions/NSURLSession+FetchData.swift:6]: That's kind of what I intended.

Comment (tonyd256) [inline roster/Extensions/NSURLSession+FetchData.swift:6]: Ok cool

File: roster/Classes/ViewControllers/EmployeesViewController.swift [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import UIKit
+
+class EmployeesViewController: UITableViewController {
+    var employees: [TBEmployee] = []
```
Comment (kaishin) [inline roster/Classes/ViewControllers/EmployeesViewController.swift:4]: I'd go for `var employees = [TBEmployee]()` here but that's just me. The `[]` kinda annoy me, but the same can be said for `()`.

Comment (gfontenot) [inline roster/Classes/ViewControllers/EmployeesViewController.swift:4]: I far prefer this syntax. It feels weird instantiating collections that way.

Comment (kaishin) [inline roster/Classes/ViewControllers/EmployeesViewController.swift:4]: > It feels weird instantiating collections that way.

Yeh, but it all depends on how you look at it.  `Type()` is standard for initialization, and `[TBEmployee]` is a type...

You can disregard.

File: roster/Extensions/TBEmployee+Gravatar.swift [hunk 6]
```diff
@@ -0,0 +1,5 @@
+extension TBEmployee {
+    var gravatarURL: NSURL {
+        return KHGravatar.URLForGravatarEmailAddress(emailAddress, defaultImageType:.ImageIdenticon, forceDefault: false, rating: .R, size: 100.0)
```
Comment (kaishin) [inline roster/Extensions/TBEmployee+Gravatar.swift]: Space after `defaultImageType`.

File: roster/Classes/ViewControllers/EmployeesViewController.swift (hunk 5 above)
Comment (tonyd256) [inline roster/Classes/ViewControllers/EmployeesViewController.swift:4]: I'm with Gordon on this one.  I like the `[]` and `[:]` to init empty arrays and dicts.  I think it aligns close with other languages where you would do `[]` or `{}` and even objc with `@[]` and `@{}`.

Comment (hyperspacemark) [inline roster/Classes/ViewControllers/EmployeesViewController.swift:4]: It sounds like you've made up your mind but I'm with Reda here. The type declaration seems unnecessary and the parentheses immediately clue me into what's actually happening.

**Comment by hyperspacemark** [comment]
One comment agreeing with Reda. Looks good.

File: roster/Classes/ViewControllers/EmployeesViewController.swift (hunk 5 above)
Comment (gfontenot) [inline roster/Classes/ViewControllers/EmployeesViewController.swift:4]: Sounds like a discussion to have in a guides PR! I'll open one up momentarily.

**Pull request merged**